### PR TITLE
SW-5318 Participant Project Species permissions

### PR DESCRIPTION
**Pull request by nickgraz**
- Adds 4 new permissions: 
  - `canCreateParticipantProjectSpecies`
  - `canDeleteParticipantProjectSpecies`
  - `canReadParticipantProjectSpecies`
  - `canUpdateParticipantProjectSpecies`

### CODE REVIEW

**Comment by nickgraz** [comment]
* **#1937** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1937?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1935** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1935?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1934** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1934?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1933** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1933?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1932** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1932?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1924** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1924?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1919** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1919?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1918** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1918?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1917** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1917?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1916** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1916?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/test/kotlin/com/terraformation/backend/customer/model/PermissionTest.kt [hunk 1]
```diff
@@ -2143,6 +2221,7 @@ internal class PermissionTest : DatabaseTest() {
     private val uncheckedMonitoringPlots = monitoringPlotIds.toMutableSet()
     private val uncheckedObservations = observationIds.toMutableSet()
     private val uncheckedOrgs = organizationIds.toMutableSet()
+    private val uncheckedParticipantProjectSpecies = participantProjectSpeciesIds.toMutableSet()
```
Comment (sgrimm) [inline src/test/kotlin/com/terraformation/backend/customer/model/PermissionTest.kt:2224]: `andNothingElse()` needs to check the remaining IDs from this set.

File: src/test/kotlin/com/terraformation/backend/db/DatabaseTest.kt [hunk 2]
```diff
@@ -2087,6 +2091,30 @@ abstract class DatabaseTest {
     return row.id!!.also { inserted.participantIds.add(it) }
   }
 
+  private var nextParticipantProjectSpeciesNumber = 1
```
Comment (sgrimm) [inline src/test/kotlin/com/terraformation/backend/db/DatabaseTest.kt]: This doesn't seem to be used.

File: src/test/kotlin/com/terraformation/backend/db/DatabaseTest.kt [hunk 3]
```diff
@@ -2242,6 +2270,7 @@ abstract class DatabaseTest {
     val observationIds = mutableListOf<ObservationId>()
     val organizationIds = mutableListOf<OrganizationId>()
     val participantIds = mutableListOf<ParticipantId>()
+    val participantProjectSpeciesIds = mutableListOf<ParticipantProjectSpeciesId>()
```
Comment (sgrimm) [inline src/test/kotlin/com/terraformation/backend/db/DatabaseTest.kt:2277]: Maybe also add `participantProjectSpeciesId` to return the most recently-inserted ID?

**Review by sgrimm — CHANGES_REQUESTED** [review]
A couple minor things to fix here, but generally looks good. Obviously if you decide to use a compound primary key on the table, this PR will need to change too.

**Review by sgrimm — APPROVED** [review]

**Comment by nickgraz** [comment]
### Merge activity

* **May 2, 11:15 AM EDT**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1917).
* **May 2, 11:16 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1917) rebased this pull request as part of a merge.
* **May 2, 11:25 AM EDT**: @nickgraz merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1917).

**Pull request merged**